tests/FileRemovalTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+use org\bovigo\vfs\vfsStream;
+use PhpCsFixer\FileRemoval;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\FileRemoval
+ */
+final class FileRemovalTest extends TestCase
```

Comment:
Whole idea of `FileRemoval` is to still remove the file even if php process crashed and destructor was not called - we do it with shutdown function.
Not sure how to test this...

Reply:
What about [`@runInSeparateProcess`](https://phpunit.readthedocs.io/en/7.1/annotations.html#runinseparateprocess)? Maybe we could have one test in a separate process and others depending on it so they are run after and can check if the file was actually removed after the first test? I have no idea whether this is even possible though :p

Reply:
How's that solution?

Reply:
I was rather thinking about creating a real file in e.g. `/tmp` and observe it without explicitly ending the PHP separate process to let PHPUnit run normally (no `exit`); and then have another test in the main process that checks that the real file does not exist anymore. Maybe this could be done using the [`@depends` annotation](https://phpunit.readthedocs.io/en/7.1/annotations.html#depends).

Reply:
Nearly got it working like that, just one thing:
Because the initial test (create file, observe file) runs in a separate process, it invokes tearDownAfterClass after it finishes. I'm using tearDownAfterClass to clean up the created file in case an assertion fails, so the files are deleted by tearDownAfterClass instead of FileRemoval. Any ideas?

Reply:
Maybe a static property like `$removeFileOnTearDown` set to `true` by default, and set to `false` in the test that creates and observes the file: this way you would know in the `tearDownAfterClass` method if you can delete the file.

Reply:
Perfect, thanks a lot!